src/gdal/ogr_file_format.cpp

```diff
@@ -1405,5 +1379,573 @@ LatLon OgrFileImport::calcAverageLatLon(OGRDataSourceH data_source)
+	po_ds = ogr::unique_datasource(OGR_Dr_CreateDataSource(
+	                                   po_driver,
+	                                   path.toLatin1(),
+	                                   nullptr));
+	if (!po_ds)
+		throw FileFormatException(tr("Failed to create dataset: %1").arg(QString::fromLatin1(CPLGetLastErrorMsg())));
+
+	// For GPX/LIBKML driver, coords have to be in EPSG:4326/WGS 1984
+	auto geo_srs = ogr::unique_srs { OSRNewSpatialReference(nullptr) };
+	OSRSetWellKnownGeogCS(geo_srs.get(), "WGS84");
+	transformation = ogr::unique_transformation { OCTNewCoordinateTransformation(map_srs.get(), geo_srs.get()) };
+
+	// Name field definition
+	o_name_field = ogr::unique_fielddefn(OGR_Fld_Create("Name", OFTString));
+	OGR_Fld_SetWidth(o_name_field.get(), 32);
+
+	// Setup style table
+	populateStyleTable();
+
+	auto is_point_object = [](const Object* object) {
+		const auto& symbol = object->getSymbol();
+		return symbol && symbol->getContainedTypes() & Symbol::Point;
+	};
+
+	auto is_text_object = [](const Object* object) {
+		const auto& symbol = object->getSymbol();
+		return symbol && symbol->getContainedTypes() & Symbol::Text;
+	};
+
+	auto is_line_object = [](const Object* object) {
+		const auto& symbol = object->getSymbol();
+		return symbol && (symbol->getType() == Symbol::Line
+		        || (symbol->getType() == Symbol::Combined && !(symbol->getContainedTypes() & Symbol::Area)));
+	};
+
+	auto is_area_object = [](const Object* object) {
+		const auto& symbol = object->getSymbol();
+		return symbol && symbol->getContainedTypes() & Symbol::Area;
+	};
+
+	if (one_layer)
+	{
+		auto layer = createLayer("Layer", wkbUnknown);
+		if (layer == nullptr)
+			throw FileFormatException(tr("Failed to create layer for objects"));
+
+		addPointsToLayer(layer, is_point_object);
+		addTextToLayer(layer, is_text_object);
+		addLinesToLayer(layer, is_line_object);
+		addAreasToLayer(layer, is_area_object);
+	}
+	else if (option(QString::fromLatin1("Per Symbol Layers")).toBool())
+	{
+		// Determine symbols in use
+		std::vector<bool> symbols_in_use;
+		map->determineSymbolsInUse(symbols_in_use);
+
+		// Add points, lines, areas in this order for driver compatability (esp GPX)
+		for (auto i = map->getNumSymbols() - 1; i >= 0; --i)
+		{
+			if (symbols_in_use[i])
+			{
+				const auto& symbol = map->getSymbol(i);
+				if (symbol->isHelperSymbol())
+					continue;
+
+				if (symbol->getType() == Symbol::Point)
+				{
+					auto layer = createLayer(QString::fromLatin1("%1_%2").arg(info.baseName(), symbol->getPlainTextName()).toLatin1(), wkbPoint);
+					if (!layer)
+					{
+						addWarning(QString::fromLatin1("Failed to export symbol %1, %2")
+						           .arg(symbol->getPlainTextName())
+						           .arg(QString::fromLatin1(CPLGetLastErrorMsg())));
+						continue;
+					}
+					addPointsToLayer(layer, [&symbol](const Object* object) {
+						const auto& sym = object->getSymbol();
+						return sym && sym->equals(symbol);
```

Comment:
Can't we simply use `sym == symbol`? 
If you really want to merge equal symbols into a single layer, you have to make sure that you don't add objects twice when looping over all symbols.
(Multiple occurrences.)

Reply:
Yes, you're correct.  I misinterpreted equals() and was thinking it was similar to QString::compare